src/frontend/komponenter/Fagsak/Vilkaar/VelgVilkaarene.tsx

```diff
@@ -0,0 +1,122 @@
+import * as React from 'react';
+import { IFagsak, IBehandling, IVedtakForBehandling, VedtakResultat } from '../../../typer/fagsak';
+import { useHistory } from 'react-router';
+import { Knapp } from 'nav-frontend-knapper';
+import { Systemtittel, Normaltekst } from 'nav-frontend-typografi';
+import { RadioPanelGruppe } from 'nav-frontend-skjema';
+import { axiosRequest } from '../../../api/axios';
+import { Ressurs, RessursStatus } from '../../../typer/ressurs';
+import AlertStripe from 'nav-frontend-alertstriper';
+
+interface IProps {
+    fagsak: IFagsak;
+}
+
+const VelgVilkaarene: React.FunctionComponent<IProps> = ({ fagsak }) => {
+    const history = useHistory();
+    const [fyltvilkår, settFyltvilkår] = React.useState(true);
+    const [feilmelding, settFeilmelding] = React.useState('');
+    const [senderInn, settSenderInn] = React.useState(false);
+
+    const aktivBehandling = fagsak.behandlinger.find(
+        (behandling: IBehandling) => behandling.aktiv === true
+    );
+
+    const aktivVedtak = aktivBehandling
+        ? aktivBehandling.vedtakForBehandling.find(
+              (vedtak: IVedtakForBehandling) => vedtak.aktiv === true
+          )
+        : undefined;
+
+    React.useEffect(() => {
+        if (aktivVedtak) {
+            settFyltvilkår(aktivVedtak.resultat == VedtakResultat.INNVILGET);
+        }
+    }, []);
+
+    if (!aktivBehandling) {
+        return (
+            <div>
+                <Normaltekst>Ingen aktiv behandling</Normaltekst>
+            </div>
+        );
+    }
+
+    return (
+        <div>
+            {feilmelding !== '' && <AlertStripe type="feil">{feilmelding}</AlertStripe>}
+            <div className={'vilkårene'}>
+                <Systemtittel children={'Inngangsvilkår'} />
+
+                <br />
+                <RadioPanelGruppe
+                    name={'inngangsvilkår'}
+                    legend={'Inngangsvikår for barnetrygd'}
+                    radios={[
+                        { label: 'Inngangsvilkårene er oppfylt', value: 'oppfylt' },
+                        { label: 'Inngangsvilkårene er ikke oppfylt', value: 'ikkeoppfylt' },
+                    ]}
+                    checked={fyltvilkår ? 'oppfylt' : 'ikkeoppfylt'}
+                    onChange={(evt: {}, value: string) => {
+                        settFyltvilkår(value === 'oppfylt');
+                    }}
+                />
+
+                <br />
+                <div className={'fastsett__navigering'}>
```

Comment:
Proposal:
Most of our steps has similar pattern:
1. navigation
2. Skjema (form)
3. Submit-button

Can you see a way to create a parent component which lets us easily create more steps?

Reply:
What I can see is that a parent component implement a common framework of each page, and the behavior of submit-button can be customized as part of the properties. The Skjema should be customized for each page so it can be the child of the parent component.

Reply:
Yes, exactly 💪